Add RTF output

tdwils opened the pull request:
Addresses #38 .

RTF has been added as a Download Type option:

![image](https://user-images.githubusercontent.com/16843423/211386823-fec3b3cb-eaf5-42b1-b23b-9ae0d360dedc.png)

Sample output:

![image](https://user-images.githubusercontent.com/16843423/211386856-2da312a1-29e7-4751-8ff3-8a7111ba654b.png)

I added a blank line between the blocks. (The blank line also now appears in the HTML and CSV, but we can remove it if necessary.)

The output from `gt` seems to have limited formatting capabilities for RTF and docx types:

- Formatting options like font family and size work for HTML output but not for RTF or docx.
- The footnotes only appear on the last page. They are not repeated on all pages.
- Page numbers can be added, but only in the upper right corner.
- Adding RTF codes (like bold: \b, \b0) are not rendered. So instead the footnotes are formatted as plain text for now:

![image](https://user-images.githubusercontent.com/16843423/211387717-f0229003-da62-4cff-b3d0-18b479f7013e.png)


We could keep this simple format and stay with the `gt` package, implementing more formatting as it’s added to the `gt` package. 

Or we could use `r2rtf` to make a nicer table now. What do you think? What other formatting would we want to implement?

Jeff-Thompson12 commented:
@tdwils I was messing around in the app with this branch and noticed a warning gets flashed up when selecting one of the larger STAN tables. For instance, I saw the screenshot below when I selected "Table 5: Demography". The table did still load in the end though.
![image](https://user-images.githubusercontent.com/85879620/211570234-a44de8e0-8c06-4989-9b1d-97dbe4905d3a.png)

tdwils commented:
> @tdwils I was messing around in the app with this branch and noticed a warning gets flashed up when selecting one of the larger STAN tables. For instance, I saw the screenshot below when I selected "Table 5: Demography". The table did still load in the end though. 

@Jeff-Thompson12  I am not able to reproduce this error. Instead, I am getting different behavior. When I start up the app and select one of the Standard Analysis Tables, nothing happens. It is only when I select a second table that a table is actually generated. But I still don't see the error you are getting. Maybe this should be a separate issue.

Jeff-Thompson12 commented:
> Maybe this should be a separate issue.

I'm not opposed to that. I have only noticed it on this branch, so I thought it might be related to the blank column being added, but I did not investigate.

tdwils commented:
> @tdwils I also output a an RTF table and the margins seemed off. Is it possible to center the table using the `gt` package? I'm assuming centering the table on the page would be desired behavior. [TableGenerator.zip](https://github.com/Biogen-Inc/tidyCDISC/files/10392735/TableGenerator.zip)

@Jeff-Thompson12  I have changed the page layout to landscape and set the table to fill up the page width.  Columns are evenly distributed.  The `gt` package currently only supports column widths that are passed as hardcoded values, so we can't pass values that are customized for each table (Issue: https://github.com/rstudio/gt/issues/641). (E.g., the first column should be the widest, and the rest should be evenly distributed)

![image](https://user-images.githubusercontent.com/16843423/212095449-3abad4a9-c0a2-401c-afa4-63d29cfee418.png)


I have changed this to draft, and I am going to go through all of the standard tables to see if there is any other custom tweaking we can do for the RTF output.

AARON-CLARK commented:
Hi @tdwils & @Jeff-Thompson12 - I know this was converted to draft, but I just wanted to share my experience:

All the standard analyses tables worked fine except for Tables 5 & 41, where I got this message:

![image](https://user-images.githubusercontent.com/17878953/214564260-70e3d932-9936-4246-9249-34ad52b1ee57.png)

> I am getting different behavior. When I start up the app and select one of the Standard Analysis Tables, nothing happens. It is only when I select a second table that a table is actually generated.

Thankfully, I'm not experiencing that! Do you still have that issue?


@tdwils: overall, great investigation of the `gt` pkg's new RTF features! I think it's shortcomings are to be expected for now, being its relatively new to `gt`. It's hard to guess when their development will catch up with something like `r2rtf`. My guess is that it will be slow (maybe a year). For now, let's see how polished we can make the outputs using `gt` before we decide to switch to something else like `r2rtf`. In general, our goal is for `tidyCDISC` to produce more "realistic" CSR-like outputs, even though we aren't aiming for submission quality right now. The goal is to increase users trust of the output by making it look similar to what they're used to reading.

> I added a blank line between the blocks. (The blank line also now appears in the HTML and CSV, but we can remove it if necessary.)

Yes, could you please remove it from the HTML? I agree it looks nice on the RTF.

> the first column should be the widest, and the rest should be evenly distributed. 

Shoot!

tdwils commented:
Hi @AARON-CLARK 

> All the standard analyses tables worked fine except for Tables 5 & 41, where I got this message:
> 
> ![image](https://user-images.githubusercontent.com/17878953/214564260-70e3d932-9936-4246-9249-34ad52b1ee57.png)
> 

I have made some updates and tested all the standard analysis tables with the CDISC data. They all generated without error.

> > I added a blank line between the blocks. (The blank line also now appears in the HTML and CSV, but we can remove it if necessary.)
> 
> Yes, could you please remove it from the HTML? I agree it looks nice on the RTF.
> 

I have changed the output so that only the RTF has a blank row between blocks.

AARON-CLARK commented on R/mod_tableGen_utils.R:
```diff
@@ -563,14 +563,57 @@ tg_guide <- cicerone::Cicerone$
+#' @param col_total A vector of column totals
+#' @param subtitle The table subtitle
+#'  
+#' @export
```
No need to `@export` right now - please remove and re-run `run_dev.R`. There is another function used in the reproducible script called `tg_gt()` that builds the gt object for the R script. We can potentially look at editing that in a future PR so that the user can create an RTF from an R script.

tdwils replied:
@AARON-CLARK  Changed `create_gt_table()` to not exported. Ready for re-review.

AARON-CLARK reviewed (changes requested):
Everything looks great @tdwils. The only comment I have regarding exporting the new function.
